### DISCUSSION TITLE
Re-add Windows support and ship builds for it

### DISCUSSION THREAD

**Discussion by feikname** [piqueserver/piqueserver#635] [Feature Requests]

**Comment by feikname** [comment]
I marked this as low priority because now Ubuntu and other distros are pretty much included inside Windows 10.

**Comment by NotAFile** [comment]
The only reason why we would want Windows support anway would be for LAN servers.

**Comment by noway** [comment]
Yep, very good point. I think Windows 10 Ubuntu tutorial would be more valuable than setting up windows builds and fixes. 

*imho*

**Comment by NotAFile** [comment]
I believe in supporting it well enough that we can launch a rudimentary vanilla LAN server from inside of OpenSpades or via some external GUI

**Comment by kmsiapps** [comment]
Mmm.. what needs to be fixed to run Piqueserver on Windows?

**Comment by NotAFile** [comment]
- [ ] compilation/setup.py
- [ ] setting the correct config dir (`%APPDATA%\piqueserver`)
- [ ] getting py2exe to make a valid executable

this is all I think

**Comment by kmsiapps** [comment]
Okay, so I succeded building setup.py by following this steps:

- Install chocolatey
- `choco install python2` and `choco  install mingw` on Powershell
- Downloaded piqueserver-master.zip from Github and extracted it
- open Powershell from extracted directory
- `pip install pyenet` (got an error when installed with pip install piqueserver)
- `pip install piqueserver` (for downloading dependencies)
- `python setup.py build -cmingw32`
- `python setup.py install`

Sometimes there's some errors when I installed python and mingw with official installer. Don't know why.

Also, It seems that it's pyenet that blocks running Piqueserver on Windows. I tested on two Windows machines, and got the same DLL error on pyenet.

**Comment by feikname** [comment]
@kmsiapps what is the error that you get when trying to install pyenet with pip on Windows?

It might some change we've made to the original pyenet that is causing it, dunno... [this is the diff](https://github.com/piqueserver/pyenet/compare/265b18ed...master) between the original pyenet and our version.

**Comment by NotAFile** [comment]
I doubt our changes are affecting it. It's a very weird fault.

It works completely fine on my w7 VM, but on KMSI's machine it works only when importing from the python shell, but not from within piqueserver. Creating a new blank file and putting a single "import enet" has not been tested yet.

**Comment by kmsiapps** [comment]
@NotAFile Well, `import enet` is not always working with python shell. It sometimes works, but sometimes raises error, just like with piqueserver. It's not about whether it's running on python shell or piqueserver script.

**Comment by NotAFile** [comment]
Interesting. Is it completely random even under identical conditions (in the same directory)

Might be some kind of race condition, which is extremely annoying.

dirty hack:
```
while True:
    try:
        import enet
        break
    except ImportError:
        pass
```

:P

**Comment by NotAFile** [comment]
So yes, this seems to be hinging on enet for now. With some luck, switching to python3 will fix this.

**Comment by NotAFile** [comment]
The move to python3 did in fact fix this.

**Comment by feikname** [comment]
@NotAFile So the issue can be closed, right? I think we forgot it open, lol.

**Comment by godwhoa** [comment]
@feikname Works well with Windows 10 + VS 2017 + upto-date setuptools. But didn't have much luck on Windows 8.

**Comment by NotAFile** [comment]
I thing this can be replaced with a "ship windows builds" thing instead? Ideally as a single exe.
